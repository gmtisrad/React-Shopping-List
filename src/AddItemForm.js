import React from 'react';

class AddItemForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputValue: '',
        }
    }

    onSubmitForm = (event) => {
        event.preventDefault();
        this.props.onAddItem(this.state.inputValue);
        this.setState({
            inputValue: '',
        })
    }

    updateInputValue(event) {
        this.setState({
            inputValue: event.target.value,
        })
    }

    render() {
        return (
            <form 
                className='add-item-form'
                onSubmit={(event) => this.onSubmitForm(event)}
            >
              <section className='input-section'>
                <input 
                    type='text' 
                    id='add-item-input' 
                    placeholder='Add Item' 
                    value={this.state.inputValue}
                    onChange={event => this.updateInputValue(event)}/>
                <input 
                    type='submit' 
                    id='add-item-submit' 
                    value='Add Item'/>
              </section>
            </form>
        )
    }
}

export default AddItemForm;