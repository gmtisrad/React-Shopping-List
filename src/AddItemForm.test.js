import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import AddItemForm from './AddItemForm.js';
import toJson from 'enzyme-to-json';

describe ('Tests for AddItemForm Component', () => {
    it('Renders without crash', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AddItemForm />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders UI as expected', () => {
        const wrapper = shallow(<AddItemForm />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})