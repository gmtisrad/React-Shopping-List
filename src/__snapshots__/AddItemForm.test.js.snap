// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for AddItemForm Component renders UI as expected 1`] = `
<form
  className="add-item-form"
>
  <section
    className="input-section"
  >
    <input
      id="add-item-input"
      placeholder="Add Item"
      type="text"
    />
    <input
      id="add-item-submit"
      type="submit"
      value="Add Item"
    />
  </section>
</form>
`;
